pkg/rpc/client/client_test.go

```diff
@@ -1,116 +1,47 @@
 package client
 
 import (
-	"context"
-	"errors"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
 	"net/http"
 	"net/http/httptest"
 	"testing"
 
 	"github.com/golang/protobuf/ptypes/timestamp"
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/encoding/protojson"
 
 	ftypes "github.com/aquasecurity/fanal/types"
 	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
 	"github.com/aquasecurity/trivy-db/pkg/utils"
 	"github.com/aquasecurity/trivy/pkg/types"
 	"github.com/aquasecurity/trivy/rpc/common"
-	"github.com/aquasecurity/trivy/rpc/scanner"
+	rpc "github.com/aquasecurity/trivy/rpc/scanner"
 )
 
-type mockScanner struct {
-	mock.Mock
-}
-
-type scanArgs struct {
-	Ctx             context.Context
-	CtxAnything     bool
-	Request         *scanner.ScanRequest
-	RequestAnything bool
-}
-
-type scanReturns struct {
-	Res *scanner.ScanResponse
-	Err error
-}
-
-type scanExpectation struct {
-	Args    scanArgs
-	Returns scanReturns
-}
-
-func (_m *mockScanner) ApplyScanExpectation(e scanExpectation) {
-	var args []interface{}
-	if e.Args.CtxAnything {
-		args = append(args, mock.Anything)
-	} else {
-		args = append(args, e.Args.Ctx)
-	}
-	if e.Args.RequestAnything {
-		args = append(args, mock.Anything)
-	} else {
-		args = append(args, e.Args.Request)
-	}
-	_m.On("Scan", args...).Return(e.Returns.Res, e.Returns.Err)
-}
-
-func (_m *mockScanner) ApplyScanExpectations(expectations []scanExpectation) {
-	for _, e := range expectations {
-		_m.ApplyScanExpectation(e)
-	}
-}
-
-// Scan provides a mock function with given fields: Ctx, Request
-func (_m *mockScanner) Scan(Ctx context.Context, Request *scanner.ScanRequest) (*scanner.ScanResponse, error) {
-	ret := _m.Called(Ctx, Request)
-
-	var r0 *scanner.ScanResponse
-	if rf, ok := ret.Get(0).(func(context.Context, *scanner.ScanRequest) *scanner.ScanResponse); ok {
-		r0 = rf(Ctx, Request)
-	} else {
-		if ret.Get(0) != nil {
-			r0 = ret.Get(0).(*scanner.ScanResponse)
-		}
-	}
-
-	var r1 error
-	if rf, ok := ret.Get(1).(func(context.Context, *scanner.ScanRequest) error); ok {
-		r1 = rf(Ctx, Request)
-	} else {
-		r1 = ret.Error(1)
-	}
-
-	return r0, r1
-}
-
 func TestScanner_Scan(t *testing.T) {
-	type fields struct {
-		customHeaders CustomHeaders
-	}
 	type args struct {
 		target   string
 		imageID  string
 		layerIDs []string
 		options  types.ScanOptions
 	}
 	tests := []struct {
-		name            string
-		fields          fields
-		args            args
-		scanExpectation scanExpectation
-		wantResults     types.Results
-		wantOS          *ftypes.OS
-		wantEosl        bool
-		wantErr         string
+		name          string
+		customHeaders http.Header
+		args          args
+		expectation   *rpc.ScanResponse
```

Comment:
We can use `interface{}` for defining an error.
https://github.com/aquasecurity/trivy/blob/04d86ece69232e64ee9cfca3c6471998fe84d4c5/pkg/rpc/client/client_test.go#L169-L175

Reply:
I've changed this block. but i'm not sure that it's correct.